ygot/diff.go

```diff
@@ -432,13 +439,14 @@ func Diff(original, modified GoStruct, opts ...DiffOpt) (*gnmipb.Notification, e
 			n.Delete = append(n.Delete, origPath.gNMIPaths...)
 		}
 	}
-
-	// Check that all paths that are in the modified struct have been examined, if
-	// not they are updates.
-	for modPath, modVal := range modLeaves {
-		if !matched[modPath] {
-			if err := appendUpdate(n, modPath, modVal); err != nil {
-				return nil, err
+	if pathOpt == nil || !pathOpt.IgnoreNewPaths {
```

Comment:
Suggest to reverse this logic -- i.e., `if pathOpt != nil && pathOpt.IgnoreNewPaths { return ... }`

Reply:
Done.